src/sentry_core.c

```diff
@@ -874,14 +874,29 @@ sentry_transaction_start(
 
     sentry_value_set_by_key(tx, "start_timestamp",
         sentry__value_new_string_owned(
-            sentry__usec_time_to_iso8601(sentry__usec_time())));
+            sentry__usec_time_to_iso8601(timestamp)));
 
     sentry__transaction_context_free(opaque_tx_cxt);
     return sentry__transaction_new(tx);
 }
 
+sentry_transaction_t *
+sentry_transaction_start(
+    sentry_transaction_context_t *opaque_tx_cxt, sentry_value_t sampling_ctx)
+{
+    return sentry__transaction_start(
+        opaque_tx_cxt, sampling_ctx, sentry__usec_time());
+}
+
+sentry_transaction_t *
+sentry_transaction_start_ts(sentry_transaction_context_t *opaque_tx_cxt,
+    sentry_value_t sampling_ctx, uint64_t timestamp)
+{
+    return sentry__transaction_start(opaque_tx_cxt, sampling_ctx, timestamp);
+}
+
 sentry_uuid_t
-sentry_transaction_finish(sentry_transaction_t *opaque_tx)
+sentry__transaction_finish(sentry_transaction_t *opaque_tx, uint64_t timestamp)
```

Comment:
You have two options here: either you implement `sentry_transaction_start()` in terms of `sentry_transaction_start_ts()` (like we did with the `_n`-functions) or you push down the implementation to a shared function that duplicates the interface wrt the `_ts` counterpart. If you do the latter, the pushed-down implementation should be static and lose the `sentry__` prefix.